src/coreclr/src/jit/importer.cpp

```diff
@@ -6393,6 +6393,15 @@ bool Compiler::impCanPInvokeInlineCallSite(BasicBlock* block)
     //   jit\jit64\ebvts\mcpp\sources2\ijw\__clrcall\vector_ctor_dtor.02\deldtor_clr.exe
     if (block->hasTryIndex())
     {
+        if (opts.jitFlags->IsSet(JitFlags::JIT_FLAG_IL_STUB) &&
+            opts.jitFlags->IsSet(JitFlags::JIT_FLAG_PUBLISH_SECRET_PARAM))
```

Comment:
Is this condition correct? 

`JIT_FLAG_PUBLISH_SECRET_PARAM` is never set in crossgen2. It means that this `if` is going to be always false in crossgen2 and this change has no effect on what crossgen2 does.

Reply:
Oops... Typo probably during the repro reforking

Reply:
Fixed. Thanks for catching this Jan

Reply:
Do we need the `JIT_FLAG_PUBLISH_SECRET_PARAM` part of the condition at all? What is not going to work if the condition is just `if (opts.jitFlags->IsSet(JitFlags::JIT_FLAG_IL_STUB)` ?

Reply:
> What is not going to work if the condition is just if (opts.jitFlags->IsSet(JitFlags::JIT_FLAG_IL_STUB) ?

In my mind, i think it should work, but I put a more conservative check to avoid an unintentional break somewhere else, but I don't know enough context outside of crossgen2 to be able to say with confidence that we should remove the check for `JIT_FLAG_PUBLISH_SECRET_PARAM`

Reply:
Some interop stubs generated at runtime are not shared and do not have `CORJIT_FLAG_PUBLISH_SECRET_PARAM`. It is not reliable to use `CORJIT_FLAG_PUBLISH_SECRET_PARAM` to tell whether you are generating stub for crossgen2. (And even if it was reliable, it would be rather cryptic to use this flag for this purpose.)

It means that you are still potentially changing the behavior for the JIT case with this change. The potential bugs are just going to be more subtle and harder to find.

Reply:
I see. Ok let me remove `CORJIT_FLAG_PUBLISH_SECRET_PARAM` and see if tests are going to start to fail in CI/outerloop. If all tests are passing, I'll merge without `CORJIT_FLAG_PUBLISH_SECRET_PARAM`. 

Sounds good?

Reply:
I think it would be most appropriate to change the condition to use `CORJIT_FLAG_USE_PINVOKE_HELPERS`.

Reply:
`CORJIT_FLAG_USE_PINVOKE_HELPERS` happens to be used by crossgen only, so you can be sure that you are not changing any of the JIT behavior.